.github/workflows/check.yaml

```diff
@@ -26,17 +26,46 @@ jobs:
       working-directory: ./src
 
   func:
-    uses: canonical/bootstack-actions/.github/workflows/func.yaml@v2
     needs: lint-unit
+    name: functional tests
+    runs-on: ${{ matrix.architecture.runs-on }}
+    timeout-minutes: 120
     strategy:
       fail-fast: false
       matrix:
-        command: ["TEST_JUJU3=1 make functional"]  # TEST_JUJU3 needed due https://github.com/openstack-charmers/zaza/blob/b22c2eed4c322f1dfc14ffb2d31e0dd18c911a40/setup.py#L47 to support Juju3+
         juju-channel: ["3.4/stable"]
-    with:
-      command: ${{ matrix.command }}
-      juju-channel: ${{ matrix.juju-channel }}
-      nested-containers: false
-      provider: "lxd"
-      python-version: "3.10"
-      timeout-minutes: 120
+        architecture:
+          - runs-on: [ubuntu-latest]  # github hosted runners are amd64
+            model-constraints: ""
+            # Ubuntu_ARM64_4C_16G_01 is the github-hosted arm64 runner we have access to.
+            # We prefer the github runners because they are smaller machines and save resources.
+            # If we have issues with it, we can switch to the larger and more numerous self-hosted options:
+            # - runs-on: [self-hosted, jammy, ARM64]
+          - runs-on: [Ubuntu_ARM64_4C_16G_01]
+            model-constraints: arch=arm64
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+      - name: Setup Juju environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          provider: "lxd"
+          juju-channel: ${{ matrix.juju-channel }}
+      - name: Install latest tox version
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install tox
```

Comment:
This is not needed, since tox is installed via action-operator. We used to re-install specific version.

Reply:
Done